src/utils.js

```diff
@@ -36,6 +37,54 @@ function emitEvent(...args) {
   }
 }
 
+/* eslint-disable no-restricted-properties */
+export function getWinDimensions(path = null) {
+  if (!windowDimensions) {
+    const top = canAccessWindowTop() ? internal.getWindowTop() : internal.getWindowSelf();
+
+    windowDimensions = {
+      screen: {
+        width: top.screen?.width,
+        height: top.screen?.height,
+        availWidth: top.screen?.availWidth,
+        availHeight: top.screen?.availHeight,
+        colorDepth: top.screen?.colorDepth,
+      },
+      innerHeight: top.innerHeight,
+      innerWidth: top.innerWidth,
+      outerWidth: top.outerWidth,
+      outerHeight: top.outerHeight,
+      visualViewport: {
+        height: top.visualViewport?.height,
+        width: top.visualViewport?.width,
+      },
+      document: {
+        documentElement: {
+          clientWidth: top.document?.documentElement?.clientWidth,
+          clientHeight: top.document?.documentElement?.clientHeight,
+          scrollTop: top.document?.documentElement?.scrollTop,
+          scrollLeft: top.document?.documentElement?.scrollLeft,
+        },
+        body: {
+          scrollTop: document.body?.scrollTop,
+          scrollLeft: document.body?.scrollLeft,
+          clientWidth: document.body?.clientWidth,
+          clientHeight: document.body?.clientHeight,
+        },
+      }
+    };
+  };
+
+  return path ? deepAccess(windowDimensions, path) : windowDimensions;
```

Comment:
why the path argument?  returning just the object and using optional chaining is (slightly) better than using `deepAccess`.